test/spec/request-test.js

```diff
@@ -1425,4 +1427,263 @@ describe('Request-Promise', function () {
+                });
+            });
+
+            cacheServer.listen(5000, function () {
+                done();
+            });
+
+        });
+
+        after(function () {
+            cacheServer.close();
+        });
+
+        beforeEach(function () {
+            // require a fresh lib
+            delete require.cache[require.resolve('../../lib/rp.js')]
+            rp = require('../../lib/rp.js')
+
+            // reset the last modified of the resources
+            for (i=0; i<resources.length; i++){
+                resources[i].headers['Last-Modified'] = moment().format(LAST_MODIFIED_FORMAT)
+            }
+        });
+
+        assertResponse = function(response, statusCode, content, lastModified, age) {
+            expect(response.statuCode).to.be.equal(statusCode)
+            expect(JSON.parse(response.body).name).to.be.equal(content)
+            expect(response.headers['last-modified']).to.be.equal(now)
+            if (!age) {
+                expect(response.headers['age']).to.be.equal(age)
+            } else {
+                expect(response.headers['age'] > 0).to.be.true
+            }
+        }
+
+        it('should cache resource : cache-control, cache option true', function (done) {
+            var options = {
+                uri: 'http://localhost:5000/resource1',
+                method: 'GET',
+                cache: true
+            }
+
+            rp(options).then(function(response){
+                assertResponse(response, 200, 'resource1', now, 0)
+                rp(options).then(function(response){
+                    assertResponse(response, 200, 'resource1', now, true)
+                    done()
+                })
+            })
+
+        });
+
+        it('should not cache resource : no cache-control, cache option true', function (done) {
+            var options = {
+                uri: 'http://localhost:5000/resource2',
+                method: 'GET',
+                cache: true
+            }
+
+            rp(options).then(function(response){
+                assertResponse(response, 200, 'resource2', now, 0)
+                rp(options).then(function(response){
+                    assertResponse(response, 200, 'resource2', now, 0)
+                    done()
+                })
+            })
+
+        });
+
+        it('should not cache resource : cache-control no-cache', function (done) {
+            var options = {
+                uri: 'http://localhost:5000/resource3',
+                method: 'GET',
+                cache: true
+            }
+
+            rp(options).then(function(response){
+                assertResponse(response, 200, 'resource3', now, 0)
+                rp(options).then(function(response){
+                    assertResponse(response, 200, 'resource3', now, 0)
+                    done()
+                })
+            })
+
+        });
+
+        it('should not cache resource : cache-control, cache option false', function (done) {
```

Comment:
Test case

Assumptions: 
- the resource returns a `cache-control` header with a positive max-age
- the cache is previously empty
- the cache option is false

results:
- the resource does not get cached after first request
- the resource that we get after second request is a fresh one